internal/app/machined/pkg/system/runner/process/process.go

```diff
@@ -127,16 +194,20 @@ func (p *processRunner) run(eventSink events.Recorder) error {
 		defer reaper.Stop(notifyCh)
 	}
 
-	if err = cmd.Start(); err != nil {
+	if err = cmdWrapper.cmd.Start(); err != nil {
 		return fmt.Errorf("error starting process: %w", err)
 	}
 
-	eventSink(events.StateRunning, "Process %s started with PID %d", p, cmd.Process.Pid)
+	if cmdWrapper.afterStart != nil {
```

Comment:
this doesn't seem to be `nil` ever, and I guess it should be called even if `Start()` fails?

Reply:
correct, fixed.